python/pyarrow/tests/test_compute.py

```diff
@@ -2032,6 +2032,11 @@ def _check_temporal_rounding(ts, values, unit):
         expected = ts.dt.round(frequency)
         np.testing.assert_array_equal(result, expected)
 
+    # Check RoundTemporalOptions defaults
+    result = pc.round_temporal(ta).to_pandas()
```

Comment:
Note that this doesn't fully test the the fixed bug, as the above already worked correctly on master. It's when specifying "some" option, that the python option class gets used, and thus their defaults. So eg doing `pc.round_temporal(ta, multiple=1)` would test it (because then it uses the default "day" of the python option class)

Reply:
Ugh, sorry I missed that. Follow-up PR: https://github.com/apache/arrow/pull/12511